provider/oci/images_test.go

```diff
@@ -0,0 +1,393 @@
+	c.Assert(ok, jc.IsTrue)
+	c.Check(int(types[1].Mem), gc.Equals, spec.Memory)
+	c.Check(int(types[1].CpuCores), gc.Equals, spec.Cpus)
+
+	spec, ok = oci.ShapeSpecs["VM.Standard1.2"]
+	c.Assert(ok, jc.IsTrue)
+	c.Check(int(types[2].Mem), gc.Equals, spec.Memory)
+	c.Check(int(types[2].CpuCores), gc.Equals, spec.Cpus)
+}
+
+func (i *imagesSuite) TestInstanceTypesNilClient(c *gc.C) {
+	_, err := oci.InstanceTypes(nil, &i.testCompartment, &i.testImageID)
+	c.Assert(err, gc.ErrorMatches, "cannot use nil client")
+}
+
+func (i *imagesSuite) TestInstanceTypesImageWithUnknownShape(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	compute := ocitesting.NewMockOCIComputeClient(ctrl)
+	defer ctrl.Finish()
+
+	request := ociCore.ListShapesRequest{
+		CompartmentId: &i.testCompartment,
+		ImageId:       &i.testImageID,
+	}
+
+	response := ociCore.ListShapesResponse{
+		Items: []ociCore.Shape{
+			ociCore.Shape{
+				Shape: makeStringPointer("IDontExistInTheOCIProviderWasProbablyAddedLaterAndThatsWhyIHopeTheyWillAddResourceDetailsToShapesAPISoWeDontNeedToMaintainAMapping"),
+			},
+			ociCore.Shape{
+				Shape: makeStringPointer("VM.Standard2.1"),
+			},
+			ociCore.Shape{
+				Shape: makeStringPointer("VM.Standard1.2"),
+			},
+		},
+	}
+
+	compute.EXPECT().ListShapes(context.Background(), request).Return(response, nil).Times(1)
+
+	types, err := oci.InstanceTypes(compute, &i.testCompartment, &i.testImageID)
+	c.Assert(err, gc.IsNil)
+	c.Check(len(types), gc.Equals, 2)
+}
+
+func (i *imagesSuite) TestNewInstanceImage(c *gc.C) {
+	image := ociCore.Image{
+		CompartmentId:          &i.testCompartment,
+		Id:                     &i.testImageID,
+		OperatingSystem:        makeStringPointer("Canonical Ubuntu"),
+		OperatingSystemVersion: makeStringPointer("14.04"),
+		DisplayName:            makeStringPointer("Canonical-Ubuntu-14.04-2018.01.11-0"),
+	}
+
+	imgType, err := oci.NewInstanceImage(image, &i.testCompartment)
+	c.Assert(err, gc.IsNil)
+	c.Check(imgType.ImageType, gc.Equals, oci.ImageTypeGeneric)
+	c.Check(imgType.Series, gc.Equals, "trusty")
+	c.Check(imgType.CompartmentId, gc.NotNil)
+	c.Check(*imgType.CompartmentId, gc.Equals, i.testCompartment)
+	c.Check(imgType.Id, gc.Equals, i.testImageID)
+}
+
+func (i *imagesSuite) TestNewInstanceImageUnknownOS(c *gc.C) {
+	image := ociCore.Image{
+		CompartmentId:          &i.testCompartment,
+		Id:                     &i.testImageID,
+		OperatingSystem:        makeStringPointer("NotKnownToJuju"),
+		OperatingSystemVersion: makeStringPointer("14.04"),
+		DisplayName:            makeStringPointer("Canonical-Ubuntu-14.04-2018.01.11-0"),
+	}
+
+	_, err := oci.NewInstanceImage(image, &i.testCompartment)
+	c.Assert(err, gc.ErrorMatches, "os NotKnownToJuju not supported")
+}
+
+func (i *imagesSuite) TestRefreshImageCache(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	compute := ocitesting.NewMockOCIComputeClient(ctrl)
+	defer ctrl.Finish()
+
+	fakeUbuntuIDSecond := "fakeUbuntu2"
```

Comment:
given you're re-using "fakeUbuntu1" in several places as well, why did this one end up as  a const but not the other one?